CoreAlgorithmPlugins/src/au/gov/asd/tac/constellation/plugins/algorithms/clustering/infomap/flow/Network.java

```diff
@@ -58,18 +57,15 @@ public Network(final Config config, final GraphReadMethods rg) {
         Arrays.fill(nodeWeights, 1);
         sumNodeWeights = rg.getVertexCount();
 
-        graphConnections = new Iterable<Connection>() {
-            @Override
-            public Iterator<Connection> iterator() {
-                if (config.connectionType == ConnectionType.LINKS) {
-                    return new LinkIterator(rg);
-                } else if (config.connectionType == ConnectionType.EDGES) {
-                    return new EdgeIterator(rg);
-                } else if (config.connectionType == ConnectionType.TRANSACTIONS) {
-                    return new TransactionIterator(rg);
-                } else {
-                    throw new IllegalStateException(String.format("Unexpected connection type %s", config.connectionType));
-                }
+        graphConnections = () -> {
+            if (config.connectionType == ConnectionType.LINKS) {
+                return new LinkIterator(rg);
+            } else if (config.connectionType == ConnectionType.EDGES) {
+                return new EdgeIterator(rg);
+            } else if (config.connectionType == ConnectionType.TRANSACTIONS) {
+                return new TransactionIterator(rg);
+            } else {
+                throw new IllegalStateException(String.format("Unexpected connection type %s", config.connectionType));
```

Comment:
For some reason I thought you couldn't throw Exceptions from Lambdas...must be dreaming